### ISSUE TITLE
Counters and Indicator for Boost, Favourite and Bookmark

### ISSUE THREAD

**Issue by bmp** [rougier/mastodon-alt#9]
When I try to boost or favourite a toot, the action is completed and the indicator becomes bold. 
However, the counter for the appropriate action does not change in the buffer. It does change when I go to another timeline (notifications, home, profile, etc.) and return to the current buffer. Is this the intended behaviour?

With respect to bookmarks, while the action is completed and I am able to see the bookmark, is it intended that the icon/indicator does not change?

**Comment by rougier** [comment]
It is the current behavior but not the intended one... I tried to fix it using the display property to update the cound but it was not working propely. I'll need to check how it is done in mastodon.el to use the same technique.

As for bookmarks, it shoudl become bold when  you bookmark. You se no change at all?

**Comment by bmp** [comment]
Unfortunately, no it doesn't become bold is it because of the theme I am using? I am using ef-themes currently.

**Comment by rougier** [comment]
Not sure. Can you try with the default theme ?

**Comment by bmp** [comment]
Doesn't look like a font issue either, tried disable-theme and then tried to bookmark. No change at all when I bookmark

**Comment by rougier** [comment]
Can you look at the face of the bookmark symbol (using describe-face when point is on it)?

**Comment by bmp** [comment]
It says 'font-lock-comment-face'

**Comment by rougier** [comment]
Could add a `(message "Toggling bookmark")` [here](https://github.com/rougier/mastodon-alt/blob/master/mastodon-alt.el#L425) and check if you get the message when you bookmark?

**Comment by csantosb** [comment]
No, there is no message when bookmarking.

**Comment by rougier** [comment]
Can you `C-h f mastodon-toot--bookmark-toot-toggle` to check whether Emacs indicates the function is advised ?

**Comment by csantosb** [comment]
No such function. I think new one is called `mastodon-toot--toggle-bookmark`

**Comment by bmp** [comment]
> No such function. I think new one is called `mastodon-toot--toggle-bookmark`

Yep, just checked with my version too, it is `mastodon-toot--toggle-bookmark` and not `mastodon-toot--bookmark-toot-toggle`

**Comment by rougier** [comment]
I see. This means the API changed. Do you want to make a PR (with both version depending which one is bound)?

**Comment by bmp** [comment]
Does this replacement for `mastodon-alt-toot--bookmark-toot-toggle` make sense?

```
(defun mastodon-alt-toot--bookmark-toot-toggle (_orig-fun)
  "Toggle bookmark for the toot at point."

    (when-let* ((inhibit-read-only t)
                (toot (mastodon-alt-tl--status-toot)))
      (when-let* ((bounds (mastodon-alt-tl--bounds 'bookmark-field)))
        (let ((bookmarked-p (not (get-char-property (car bounds) 'bookmarked-p))))
          ;; Replace 'mastodon-toot--bookmark-toot-toggle' with 'mastodon-toot--toggle-bookmark' if the former doesn't exist
          (let ((toggle-function (if (functionp 'mastodon-toot--bookmark-toot-toggle)
                                     'mastodon-toot--bookmark-toot-toggle
                                   'mastodon-toot--toggle-bookmark)))
            (funcall toggle-function))  ; Call the appropriate toggle function
          (add-text-properties (car bounds) (cdr bounds)
               `(bookmarked-p ,bookmarked-p
                 face ,(mastodon-alt-tl--status-face bookmarked-p 0)))))))
```

Apologies, my elisp and emacs knowledge is limited to non-existent.

**Comment by rougier** [comment]
It might be even simpler than that. If you look at [line 899](https://github.com/rougier/mastodon-alt/blob/master/mastodon-alt.el#L899), this where I "advised" the original bookmark function. This means I tell emacs that anyime the advised function callled, call my function instead. so maybe only changing this line might be enough if you replace the old mastodon dunction name with the new one (and thus, no need to change the mastondon-alt function name since we can call it wathever name we want. But using the original name can help.)

There is also a code a few line after to remove the advice.
